Tests/MoSoAnalyticsTests/GleanBaseTrackerTests.swift

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Glean
+import MoSoCore
+import XCTest
+@testable import MoSoAnalytics
+
+final class GleanBaseTrackerTests: XCTestCase {
+    override func setUp() {
+        Glean.shared.resetGlean(clearStores: true)
+    }
+
+    func testImpression() {
+        // Given
+        let user = MoSoUser(username: "example@username.org", identifier: "myNickname")
+        let tracker = GleanBaseTracker(session: MoSoSession(user: user))
+        // When
+        tracker.trackImpression(
+            postID: "Post001",
+            recommendationID: "Rec001",
+            additionalInfo: "Some additional info",
+            uiIdentifier: "example.event.impression"
+        )
+        // Then
+        let events = GleanMetrics.Ui.impression.testGetValue()!
+        let event = events[0]
+
+        XCTAssertEqual(1, events.count)
+        XCTAssertEqual(event.category, "ui")
+        XCTAssertEqual(event.name, "impression")
+        XCTAssertEqual(event.extra!["mastodon_account_handle"], "example@username.org")
+        XCTAssertEqual(event.extra!["mastodon_account_id"], "myNickname")
+        XCTAssertEqual(event.extra!["mastodon_status_id"], "Post001")
+        XCTAssertEqual(event.extra!["recommendation_id"], "Rec001")
+        XCTAssertEqual(event.extra!["ui_identifier"], "example.event.impression")
+        XCTAssertEqual(event.extra!["ui_additional_detail"], "Some additional info")
+    }
+
+    func testTrackMultipleImpressions() {
+        // Given
+        let user = MoSoUser(username: "example@username.org", identifier: "myNickname")
+        let tracker = GleanBaseTracker(session: MoSoSession(user: user))
+        // When
+        tracker.trackImpression(
+            postID: "Post001",
+            recommendationID: "Rec001",
+            additionalInfo: "Some additional info",
+            uiIdentifier: "example.event.impression"
+        )
+
+        tracker.trackImpression(
+            postID: "Post002",
+            recommendationID: "Rec002",
+            additionalInfo: "Some additional info",
+            uiIdentifier: "example.event.impression"
+        )
+
+        tracker.trackImpression(
+            postID: "Post003",
+            recommendationID: "Rec003",
+            additionalInfo: "Some additional info",
+            uiIdentifier: "example.event.impression"
+        )
+
+        // Then
+        let events = GleanMetrics.Ui.impression.testGetValue()!
```

Comment:
Sometimes I like expanding on just a count check, checking a very small subset (read: 1) of other details. For example, here, it might be worth at least checking that `event[1].postID` is `Post002", `event[2].postID == "Post002". This just helps me better understand, from a test, that we're not repeatedly sending one event somehow, and that each event is distinct. 

If you decide to do so, this comment would also apply to other tests.

Reply:
agreed! I see benefit to check all three instances

Reply:
Done!